src/context/concrete_base.rs

```diff
@@ -185,6 +190,27 @@ impl ConcreteBaseAudioContext {
             LISTENER_PARAM_IDS.end,
         );
 
+        // (?) only for online context
+        if event_channel.is_some() {
+            // init event loop
+            let event_channel = event_channel.unwrap();
+            let event_handlers = base.inner.event_handlers.clone();
+            //
+            std::thread::spawn(move || loop {
+                // (?) this thread is dedicated to event so we can block
+                if let Ok(message) = event_channel.recv() {
+                    let handlers = event_handlers.lock().unwrap();
+                    // find EventHandlerInfos that match messsage
+                    if let Some(infos) = handlers.iter().find(|item| {
+                        item.node_id == message.node_id && item.event_type == message.event_type
+                    }) {
+                        (infos.callback)();
```

Comment:
Should we drop the handler at this point? Or can it fire multiple times?

Reply:
I think for context events (e.g. `onSinkChange`) we should not drop the handlers here because they might be triggered several times, my idea was to remove them when the node is dropped on the render thread.

Also the e.g. `AudioBufferSourceNode.onended` signature should probably be `Option<callback>` so that we can remove the handler manually in user code by passing `None`